scripts/generate

```diff
@@ -108,6 +109,30 @@ def ('fromJust')
     ([$.Maybe (a), a])
     (maybe_ (() => { throw new Error ('fromJust applied to Nothing'); }) (I));
 
+//    replace :: RegExp -> String -> String -> String
+const replace =
+def ('replace')
+    ({})
+    ([$.RegExp, $.String, $.String, $.String])
+    (pattern => replacement => text => text.replace (pattern, replacement));
+
+//    replace_ :: RegExp -> (Array (Maybe String) -> String) -> String -> String
+const replace_ =
+def ('replace_')
+    ({})
+    ([$.RegExp,
+      $.Fn ($.Array ($.Maybe ($.String))) ($.String),
+      $.String,
+      $.String])
+    (pattern => substitute => text =>
+       text.replace (pattern, (...args) =>
+         substitute (map (group => group == null ? Nothing : Just (group))
+                         (args.slice (1, -2)))
+       ));
+
+//    replace__ :: RegExp -> (Array String -> String) -> String -> String
+const replace__ = flip (compose (flip (replace_)) (contramap (justs)));
```

Comment:
This function is more convenient than `replace_` when the given pattern contains no *optional* capturing groups.